Mergin/validation.py

```diff
@@ -193,7 +200,15 @@ def check_offline(self):
                 # might be vector tiles - no provider name
                 continue
             if dp_name in QGIS_NET_PROVIDERS + QGIS_DB_PROVIDERS:
-                w.items.append(layer.name())
+                # raster mbtiles are loaded using WMS provider, so we need to
+                # check whether this is a local file or remote service
+                if dp_name.lower() == "wms":
+                    uri = QgsProviderRegistry.instance().decodeUri("wms", layer.source())
+                    is_local = os.path.isfile(uri["path"]) if "path" in uri else False
+                    if not is_local:
+                        w.items.append(layer.name())
+                else:
+                    w.items.append(layer.name())
```

Comment:
I think we should handle also vector tiles MBTiles here... from the code it looks like that vector tiles MBTiles layer would always get flagged as not being offline, or never get flagged - depending on whether the data provider is present. For vector tiles, we should rather check the layer type, and then check if URI contains `type=mbtiles` (and while at it, we could also allow `type=vtpk` as offline use)

Reply:
Should be addressed

Reply:
How about the code few lines above... isn't `continue` going to skip the online/offline checks?
```python
            try:
                dp_name = layer.dataProvider().name()
            except AttributeError:
                # might be vector tiles - no provider name
                continue
```

Reply:
They do... in QGIS >= 3.22... but in earlier versions (since introduction of vector tiles in 3.14) they did not have providers assigned.